templates/components/search/query_builder/criteria.html.twig

```diff
@@ -36,6 +36,15 @@
 {% set add_class = add_class|default('') %}
 {% set normalized_itemtype = itemtype|replace({'\\': ''}) %}
 
+{% if mainform is defined and mainform %}
+    {% set item = itemtype != 'AllAssets' ? call('getItemForItemtype', [itemtype]) : null %}
+    {% if item is not null and item.maybeDeleted() %}
+        <input type="hidden" id="is_deleted" name="is_deleted" value="{{ p['is_deleted'] }}">
+    {% endif %}
+    <input type="hidden" id="as_map" name="as_map" value="{{ p['as_map'] }}">
+    <input type="hidden" id="browse" name="browse" value="{{ p['browse'] }}">
+    <input type="hidden" id="unpublished" name="unpublished" value="{{ p['unpublished'] }}">
+{% endif %}
```

Comment:
Why not moving this directly right after the form opening, in `templates/components/search/query_builder/main.html.twig`? IMHO, it would ensure that these hidden fields are always present in every search form, and are not duplicated.